### PR TITLE
ipc: various fixes for i3 compat

### LINKED ISSUE #1386: ipc: harmonize `get_tree` with i3
The tree contains most of the information but does not respect the latest i3 spec: https://i3wm.org/docs/ipc.html#_tree_reply

* no `focus` array (might be difficult?)
* in i3 responses, all of these fields are mandatory for all nodes (even `root` and `output`), whereas sway adds them selectively. In many cases, a default value of `null` or `[]` should suffice.

These issues may hinder interoperability with some existing ipc libraries.

### PR DESCRIPTION
**Pull request by lbonn**
Fixes #1386 + other inconsistencies with i3's behavior.

The goal was to be able to use [focus-last.py](https://github.com/acrisci/i3ipc-python/blob/master/examples/focus-last.py) (and [my rust version](https://github.com/lbonn/i3-focus-last)) with sway. It seems to work now, if permissions are set up correctly.

The program is not that trivial and uses window events, `get_tree` and commands with criteria: this was a good way to test the ipc implementation.

Major highlight: 3 criteria missing in sway, taken from i3

### CODE REVIEW

File: sway/criteria.c [hunk 1]
```diff
@@ -137,21 +144,35 @@ static char *crit_tokens(int *argc, char ***buf, const char * const criteria_str
...
-				"Names must be unquoted, values must be quoted");
+		} else if (*argc % 2 == 0 && *head != ' ') {
+			// parse unquoted values
+			if (!valp) {
+				quoted = false;
+				valp = head;  // value starts here
+			}
+		} else if (valp && !quoted && *head == ' ') {
+			// value ends here
+			argv[(*argc)++] = valp;
```
Comment (SirCmpwn) [inline sway/criteria.c]: Can you split this on to two lines? I don't like the dereference-increment operation going on here.

File: sway/criteria.c [hunk 2]
```diff
@@ -137,21 +144,35 @@ static char *crit_tokens(int *argc, char ***buf, const char * const criteria_str
...
+			*head = '\0';
+			valp = NULL;
+			namep = head + 1;
 		}
 		head++;
 	}
+
+	// catch last unquoted value if needed
+	if (valp && !quoted && !*head) {
+		argv[(*argc)++] = valp;
```
Comment (SirCmpwn) [inline sway/criteria.c]: Ditto

File: sway/criteria.c (hunk 1 above)
Comment (lbonn) [reply sway/criteria.c]: Yes, I agree. I picked the existing style of the function.

**Comment by SirCmpwn** [comment]
Thanks!

**Pull request merged**